### DISCUSSION TITLE
Firmware updates on battery/sleepy devices

### DISCUSSION THREAD

**Discussion by Botched1** [zwave-js/zwave-js-ui#2570] [General]
Dumb question alert @robertsLando !

For sleepy end devices, I know normally when doing a firmware update you start the update, then immediately wake the device up so the update actually starts.

What happens if you don't manually wake the device though? Does the firmware update queue? Does it just fail after a timeout? Other?

I ask because I have about 20 battery/sleepy devices I need to update firmware on, but want to be lazy and not go physically wake them as some are in awkward locations. So even if I had to do 1 at a time and wait for the 12h wake-up for the update to happen, I would rather do that then get out the ladder, get in the 110F+ attic, etc.

**Comment by robertsLando** [comment]
> What happens if you don't manually wake the device though? Does the firmware update queue? Does it just fail after a timeout? Other?

It fails after a timeout, I think the timeout is about 1 minute

**Reply by Botched1** [reply]
Thanks for confirming! I guess I need to not be lazy, and go around to the devices then. Wish you could queue firmware updates for sleepy devices, but that is probably an edge case that isn't worth programming in.

**Reply by robertsLando** [reply]
@Botched1 when you press check you only get available updates, it requires you to wake up the device in order to queue actual fw informations (that for some reason could be changed while it was sleeping). So even if we queue that there would be no reason to queue for too long as then a user action is required anyway

**Reply by Botched1** [reply]
Sorry, I should have been more descriptive...

I have the firmware from the vendor, and initiate the firmware update manually (Click device, Advanced, Firmware Update, Begin) - I'm not using the new built in firmware checker for these devices (which just says none available anyway).

So in a "what would perfect look like to me", I would be able to begin the firmware update process as mentioned above, and if device was asleep as soon as the system gets the wake-up message the firmware update then begins before it puts the sleepy device back to sleep/it times out and goes back to sleep on its own.

Again, not a big "need" for most though, so likely not worth it.

**Reply by robertsLando** [reply]
It already works like that so, there is no timeout for fw updates inited in that way, it just waits indeterminately until device wakes up AFAIK

@AlCalzone

**Reply by AlCalzone** [reply]
Correct. The manual update process where you provide a file just queues the message that initiates the update. I advise against queueing many updates at once though, multiple active updates at once are not fun.

**Reply by Botched1** [reply]
I didn't know that it would queue. 

Thanks to both of you - your explanation/clarification was really helpful!
